### PR TITLE
TST: test_sql: properly drop tables with names that need to be quoted

### LINKED ISSUE #10255: to_sql : MySQL identifier cannot be entirely numeric
Reported at StackOverflow: http://stackoverflow.com/questions/30586292/valueerror-mysql-identifier-cannot-be-entirely-numeric-with-pandas-to-sql

MySQL supports numeric identifiers (if quoted) (http://dev.mysql.com/doc/refman/5.0/en/identifiers.html) 
@artemyk Was there a reason we explicitely disallowed this? (https://github.com/pydata/pandas/blob/v0.16.1/pandas/io/sql.py#L1287)

### PR DESCRIPTION
**Pull request by artemyk**
Closes #10255 .

Rewrote some of the code in `test_sql.py` so that it now properly drops tables with names that need to be quoted.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@artemyk Thanks! 

Can you add a small note in the 0.16.2 whatsnew bugfix section?

And for the rewriting of the tests, I was thinking it would maybe be better to keep this for a separate PR if you don't mind? As this would give rahter some conflicts with the connection PR, and I am already slow with reviewing that one. So I would like to first get that one in

**Comment by artemyk** [comment]
@jorisvandenbossche OK, I will submit another PR.

**Comment by artemyk** [comment]
@jorisvandenbossche Ping.  This makes the testing code a bit more robust by making sure all tables are dropped after every test.

**Comment by jorisvandenbossche** [comment]
Whoops, forgot this one. Looking at it now

File: pandas/io/tests/test_sql.py [hunk 1]
```diff
@@ -1435,7 +1485,7 @@ def test_transactions(self):
         raise nose.SkipTest("Nested transactions rollbacks don't work with Pandas")
 
 
-class _TestSQLiteAlchemy(object):
+class _TestSQLiteAlchemy(SQLAlchemyMixIn):
```
Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: Are these ones needed to inherit from the mixin? As `_TestSQLAlchemy` already does inherit from that

File: pandas/io/tests/test_sql.py [hunk 2]
```diff
@@ -1963,14 +1977,10 @@ def test_illegal_names(self):
         for ndx, ok_name in enumerate(['99beginswithnumber','12345']):
             df.to_sql(ok_name, self.conn, flavor=self.flavor, index=False,
                       if_exists='replace')
-            self.conn.cursor().execute("DROP TABLE `%s`" % ok_name)
-            self.conn.commit()
             df2 = DataFrame([[1, 2], [3, 4]], columns=['a', ok_name])
             c_tbl = 'test_ok_col_name%d'%ndx
```
Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: This line is not used anymore?

File: pandas/io/tests/test_sql.py (hunk 1 above)
Comment (artemyk) [inline pandas/io/tests/test_sql.py]: Good point --- its not needed. Removed.

File: pandas/io/tests/test_sql.py (hunk 2 above)
Comment (artemyk) [inline pandas/io/tests/test_sql.py]: It's not needed as its done automatically in `tearDown`

Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: But I mean the variable `c_tbl` which is a string, and not used anymore? (the tables are indeed dropped in teardown)

Comment (artemyk) [inline pandas/io/tests/test_sql.py]: `c_tbl` is used to create a table w. a weird column name (see the lines after its defined)

Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: if I interpret the diff correctly, you replaced `c_tbl` by ``'test_ok_col_name'?

Comment (artemyk) [inline pandas/io/tests/test_sql.py]: Whoops, was looking at the wrong line.  You're right --- now fixed.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**